contracts/facets/ThirdPartyFacet.sol

```diff
@@ -91,117 +89,124 @@ contract ThirdPartyFacet {
     }
 
     /**
-     * @notice Accuse archaeologoists of bad behaviour, by providing proof of leaked
-     * unencrypted shards before a sarcophagus is ready to be unwrapped. The minumum
-     * number of shards required to unwrap the sarcophagus should be provided for a
-     * a successful accusal.
-     * of the cursed bonds of the archs back to them, and un-curses their bonds.
+     * @notice Accuse one or more archaeologists of leaking keyshares by submitting the hashes of the leaked shares
+     * If the archaeologists responsible for those shares haven't already been accused, their locked bond will be
+     * split between the embalmer and the supplied payment address and digging fees allocated for those archaeologists will be refunded to the embalmer
+     *
+     * If k or more archaeologists are accused over the lifetime of a sarcophagus, the sarcophagus state will be updated to Accused
+     * and bonds for all remaining archaeologists will be freed
+     *
      * @param sarcoId The identifier of the sarcophagus having leaked shards
-     * @param unencryptedShardHashes At least 'm' unencrypted shard hashes as proof of bad behaviour
+     * @param unencryptedShardHashes hashes of the leaked keyshares
      * @param paymentAddress the address to which rewards should be sent if successful
      */
     function accuse(
         bytes32 sarcoId,
         bytes32[] memory unencryptedShardHashes,
```

Comment:
What is the purpose for being able to accuse multiple archs in one function? Is it just to save on transaction costs if an accuser has multiple accusals to make?

Reply:
savings on transaction cost and also if we required accusals to be one at a time, we'd be limiting rewards to the bonds for k archaeologists. With the array, an accuser could get a payout for all n if they'd all leaked their keyshare but with individual calls, the remaining n-k would be freed as soon as the sarcophagus was deemed compromised.